scripts/audit_main_upload_buckets.py

```diff
@@ -0,0 +1,620 @@
+    'CPG261792',  # rdp-kidney misformated fastq - https://batch.hail.populationgenomics.org.au/batches/378736/jobs/43
+    # acute care fasq parsing errors https://batch.hail.populationgenomics.org.au/batches/379303/jobs/24
+    'CPG259150',
+    'CPG258814',
+    'CPG258137',
+    'CPG258111',
+    'CPG258012',
+    # ohmr4 cram parsing in align issues
+    'CPG261339',
+    'CPG261347',
+    # IBMDX truncated sample? https://batch.hail.populationgenomics.org.au/batches/422181/jobs/99
+    'CPG265876',
+]
+
+AAPI = AnalysisApi()
+PAPI = ParticipantApi()
+SAPI = SampleApi()
+SEQAPI = SequenceApi()
+CLIENT = storage.Client()
+
+
+def get_bucket_subdirs_to_search(paths: list[str]) -> defaultdict[str, list]:
+    """ "
+    Takes a list of paths and extracts the bucket name and subdirectory, returning all unique pairs
+    of buckets/subdirectories
+    """
+    buckets_subdirs_to_search = defaultdict(list)  # type: defaultdict[str,list]
+    for path in paths:
+        try:
+            pc = get_path_components_from_gcp_path(path)
+        except ValueError:
+            logging.warning(f'{path} invalid')
+            continue
+        bucket = pc['bucket']
+        subdir = pc['suffix']
+        if subdir and subdir not in buckets_subdirs_to_search[bucket]:
+            buckets_subdirs_to_search[bucket].append(subdir)
+
+    return buckets_subdirs_to_search
+
+
+def get_paths_for_subdir(bucket_name, subdirectory, file_extension):
+    """Iterate through a gcp bucket/subdir and get all the blobs with the specified file extension(s)"""
+    files_in_bucket_subdir = []
+    for blob in CLIENT.list_blobs(bucket_name, prefix=subdirectory, delimiter='/'):
+        # Check if file ends with specified analysis type
+        if not blob.name.endswith(file_extension):
+            continue
+        files_in_bucket_subdir.append(f'gs://{bucket_name}/{blob.name}')
+
+    return files_in_bucket_subdir
+
+
+def find_files_in_buckets_subdirs(
+    buckets_subdirs: defaultdict[str, list], file_types: list[str]
+):
+    """
+    Takes a list of (bucket,subdirectory) tuples and finds all the files contained in that directory
+    with filetypes defined with an input list
+    """
+    files_in_bucket = []
+    for bucket, subdirs in buckets_subdirs.items():
+        for subdir in subdirs:
+            # matrixtable / hailtable subdirectories should not appear in main-upload buckets,
+            # but handle them just in case. They are too large to search
+            if '.mt' in subdir or '.ht' in subdir:
+                continue
+            files_in_bucket.extend(get_paths_for_subdir(bucket, subdir, file_types))
+
+    return files_in_bucket
+
+
+def find_sequence_files_in_bucket(bucket_name: str) -> list[str]:
+    """Gets all the gs paths to fastq files in the projects main-upload bucket"""
+    sequence_paths = []
+    for blob in CLIENT.list_blobs(bucket_name, prefix=''):
+        if blob.name.endswith(FASTQ_EXTENSIONS):
```

Comment:
We get non-fastq sequencing files from collaborators, it might be worth creating a GENOMIC_READS_EXTENSIONS (with .bam, .cram, .fastq... extensions)